letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -579,11 +676,12 @@ def enhance(self, domain, enhancement, options=None):
         try:
             return self._enhance_func[enhancement](
                 self.choose_vhost(domain), options)
-        except ValueError:
+        except KeyError:
```

Comment:
Are you sure `ValueError` cannot be raised here any more?

Please make the try block smaller:
```python
try:
    func = self._enhance_func[enhancement]
except KeyError:
   raise errors.PluginError("Unsupported ...")

func(self.choose_vhost...)
```